MetaMask/metamask-extension
# Feat/19128/update text field search to ts

## **Description**
- Update `TextFieldSearch` to TypeScript 
- Add `deprecated` folder inside for JS version of `TextFieldSearch` with a deprecation notion
- Update any production code to use the deprecated version (updates to use TS version and remove deprecation in separate PR)


## **Related issues**

Fixes: #19128 

## **Manual testing steps**

1. Run locally 
2. [Visit Storybook](http://localhost:6006/?path=/docs/components-componentlibrary-textfieldsearch--docs)
3. Type in the component and use clear button to confirm it is working (check inspect panel for errors) 
4. Visit [Icon component story](http://localhost:6006/?path=/docs/components-componentlibrary-icon--docs) to view TextFieldSearch being used as the search for Icons


## **Screenshots/Recordings**



### **Before**



### **After**


Video is both before (current storybook build) and after (running locally)
https://www.loom.com/share/07ac5a8026b843a3ae0270f380ccd623

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
